hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/AsyncHazelcastWriterP.java

```diff
@@ -158,4 +164,32 @@ private boolean asyncCallsDone() {
         checkError();
         return allWritten;
     }
+
+    static class PartitionInfo {
```

Comment:
would be better to call this PartitionService, and then you can initialize it in `init()` automatically and allow it to be used from subclasses automatically.